internal/humanitec/convert.go

```diff
@@ -157,36 +162,62 @@ func ConvertSpec(name, envID string, spec *score.WorkloadSpec, ext *extensions.H
 		"spec":    workloadSpec,
 	}
 
-	var externals = map[string]interface{}{}
+	var externals = make(map[string]interface{})
+	var shared = make([]humanitec.UpdateAction, 0)
 	for name, res := range spec.Resources {
-		if meta, exists := ext.Resources[name]; !exists || meta.Scope == "" || meta.Scope == "external" {
-			if res.Type != "service" && res.Type != "environment" {
-				externals[name] = map[string]interface{}{
+		switch res.Type {
+
+		case "service", "environment":
+			continue
+
+		default:
+			resId, hasAnnotation := res.Metadata.Annotations[AnnotationLabelResourceId]
+			if resId == "" {
+				resId = fmt.Sprintf("externals.%s", name)
+			}
+
+			// DEPRECATED: Should use resource annotations instead
+			if meta, hasMeta := ext.Resources[name]; hasMeta {
+				log.Printf("Warning: Extensions for resources has been deprecated. Use Score resource annotations instead. Extensions are stil configured for '%s'.\n", name)
+				if !hasAnnotation && (meta.Scope == "" || meta.Scope == "externals") {
+					resId = fmt.Sprintf("externals.%s", name)
+				} else if !hasAnnotation && meta.Scope == "shared" {
+					resId = fmt.Sprintf("shared.%s", name)
+				}
+			}
+			// END (DEPRECATED)
+
+			if strings.HasPrefix(resId, "externals.") {
```

Comment:
It is also possible to reference resources in other workloads - e.g. `modules.other-workload.externals.my-res`
This case would fail here.

This does not require explicit addition of a resource to the delta, but it should be allowed.

Reply:
We are adding a new feature here. This logic was not supported with the extensions file mechanism.
Let's create a separate issue and describe possible cases, requirements, and app behaviour there.

Reply:
This third case is as follows:
the res-id can have the format:
```
modules.{workloadId}.externals.{resId}
```
In this case, there are 2 outcomes:
1. if `{workloadId}` matches the current workload, treat as `externals.{resId}`
2. if `{workloadId}` does not match the current workload, just re-render the placeholder as having the value of the annotation (i.e. don't add anything to `externals` or `shared` properties.

Reply:
`workload-id` is unknown to score. what do we compare with?

Reply:
Are we to support short versions if the same res-id? E.g. `external.{resId}`? What about app-shared res-ids? How would they look like?